### PR TITLE
feat: the G1 curve should support element conversion ( PROOF-666 )

### PR DESCRIPTION
**Pull request by jacobtrombetta**
# Rationale for this change
In order to support multi-scalar multiplication, elements on the G1 curve need to be convertible between affine and projective coordinates. Commitment results from multi-scalar multiplication will be output as affine coordinates, which a library such as arkworks can handle.

# What changes are included in this PR?
- A `conversion_utility` component is added to the `curve_g1/type` package to handle conversion from projective->affine and affine->projective coordinates.
- The G1 `element_affine` component's infinity parameter is now a `uint8_t` type.
- The G1 `element_affine` component now defines the `identity`, similar to the `element_p2` component.
  - The `identity` component has been removed from the `curve_g1\properties` package.
- A generic `cmov` component is added to the `base/num` package that can move integral values.

# Are these changes tested?
Yes

### CODE REVIEW

**Review by rnburn — CHANGES_REQUESTED** [review]
Please match existing conventions: https://github.com/spaceandtimelabs/blitzar/blob/main/sxt/curve21/type/conversion_utility.h

File: sxt/field12/operation/cmov.cc [hunk 1]
```diff
@@ -28,4 +28,14 @@ CUDA_CALLABLE void cmov(f12t::element& f, const f12t::element& g, unsigned int b
     f[i] = f[i] ^ (mask & (f[i] ^ g[i]));
   }
 }
+
+//--------------------------------------------------------------------------------------------------
+// cmov
+//--------------------------------------------------------------------------------------------------
+CUDA_CALLABLE void cmov(uint8_t& f, const uint8_t g, unsigned int b) noexcept {
```
Comment (rnburn) [inline sxt/field12/operation/cmov.cc:35]: Please move this version of cmov into the base/num package.

Code that isn't field-specific and could potentially have reuse in other contexts should be in a more basic package.

Comment (jacobtrombetta) [reply sxt/field12/operation/cmov.cc]: Great idea. Moved and made a template accepting only integral types.

Comment (rnburn) [reply sxt/field12/operation/cmov.cc]: > The cmov component in the base field, field12, can now move uint8_t values

Please update description.

Comment (jacobtrombetta) [reply sxt/field12/operation/cmov.cc]: Good catch, updated.

**Review by rnburn — APPROVED** [review]

**Pull request merged**